src/clj/rems/api/applications_v2.clj

```diff
@@ -0,0 +1,869 @@
+             :workflow/id (:workflow/id event)
+             :workflow/type (:workflow/type event)
+             ;; TODO: or would :workflow.dynamic/state be more appropriate?
+             :workflow/state :rems.workflow.dynamic/draft ; TODO: other workflows
+             :workflow.dynamic/handlers (:workflow.dynamic/handlers event))
+      (set-role-permissions {:applicant #{::dynamic/add-member
+                                          ::dynamic/save-draft
+                                          ::dynamic/submit}})))
+
+(defn- set-accepted-licences [licenses accepted-licenses]
+  (map (fn [license]
+         (assoc license :license/accepted (contains? accepted-licenses (:license/id license))))
+       licenses))
+
+(defmethod application-view-specific :application.event/draft-saved
+  [application event]
+  (-> application
+      (assoc :application/modified (:event/time event)
+             :form/fields (map (fn [[field-id value]]
+                                 {:field/id field-id
+                                  :field/value value})
+                               (:application/field-values event)))
+      (update :application/licenses set-accepted-licences (:application/accepted-licenses event))))
+
+(defmethod application-view-specific :application.event/member-added
+  [application event]
+  application)
+
+(defmethod application-view-specific :application.event/submitted
+  [application event]
+  (-> application
+      (assoc :workflow/state ::dynamic/submitted)
+      (set-role-permissions {:applicant #{::dynamic/add-member}
+                             :handler #{::dynamic/approve
+                                        ::dynamic/reject
+                                        ::dynamic/return
+                                        ::dynamic/request-decision
+                                        ::dynamic/request-comment}})))
+
+(defmethod application-view-specific :application.event/returned
+  [application event]
+  application)
+
+(defmethod application-view-specific :application.event/comment-requested
+  [application event]
+  application)
+
+(defmethod application-view-specific :application.event/commented
+  [application event]
+  application)
+
+(defmethod application-view-specific :application.event/decision-requested
+  [application event]
+  application)
+
+(defmethod application-view-specific :application.event/decided
+  [application event]
+  application)
+
+(defmethod application-view-specific :application.event/approved
+  [application event]
+  application)
+
+(defmethod application-view-specific :application.event/rejected
+  [application event]
+  application)
+
+(defmethod application-view-specific :application.event/closed
+  [application event]
+  application)
+
+(deftest test-application-view-specific
+  (testing "supports all event types"
+    (is (= (set (keys dynamic/event-schemas))
+           (set (keys (methods application-view-specific)))))))
+
+
+(defn- log-event [events event]
```

Comment:
Why is this called `log-event`? There could be a better name or docstring?

Reply:
Got name suggestions? `append-to-event-history`?

Reply:
I don't understand why this function exists so I can't really think of a better name.